test/gurka/test_route_summary.cc

```diff
@@ -131,3 +121,98 @@ TEST(TestRouteSummary, GetSummary) {
 
   filesystem::remove_all(workdir);
 }
+
+TEST(TestRouteSummary, DupSummaryFix) {
+  const std::string ascii_map = R"(
+                   G-------------------H                I
+                  /                     \              / \
+      A----------B                       E------------F   J
+                 |                       |             \ /
+                 |                       |              K
+                 |                       |
+                 |                       |
+                 |                       |
+                 C-----------------------D
+    )";
+
+  const gurka::ways ways = {
+      {"AB", {{"highway", "primary"}, {"name", "RT 1"}}},
+      {"BGHE", {{"highway", "motorway"}, {"name", "RT 1"}}},
+      {"EF", {{"highway", "primary"}, {"name", "RT 2"}}},
+      {"FIJ", {{"highway", "motorway"}, {"name", "RT 3"}}},
+      {"BCDE", {{"highway", "primary"}, {"name", "RT 2"}}},
+      {"FKJ", {{"highway", "primary"}, {"name", "RT 4"}, {"access", "no"}, {"foot", "yes"}}},
+  };
+
+  const auto node_layout = gurka::detail::map_to_coordinates(ascii_map, 100, PointLL{5.108, 52.01});
+
+  std::unordered_map<std::string, std::string> config_map =
+      {{"mjolnir.data_processing.use_direction_on_ways", "true"},
+       {"mjolnir.admin", VALHALLA_SOURCE_DIR "test/data/netherlands_admin.sqlite"}};
+
+  std::string workdir = "test/data/gurka_test_dup_summary_fix";
+  valhalla::gurka::map map = gurka::buildtiles(node_layout, ways, {}, {}, workdir, config_map);
+
+  map.nodes = node_layout;
+
+  if (!filesystem::exists(workdir)) {
```

Comment:
Unrelated to your change: Just curious, could/should this be handled by the gurka framework? I don't immediately see anything specific to your test that requires this which makes me wonder if all gurka tests need this?

Reply:
gurka already does do this. i believe in the admin tests where we need to do stuff outside of gurka we needed to manually do stuf like this but calling `buildtiles` certainly does this as mjolnir will create the dir where it needs to put tiles etc

Reply:
I will fix this, thanks.